lib/core/lib/ops/validation.js

```diff
@@ -4,14 +4,39 @@ const Joi = require('joi')
 
 const requiredString = Joi.string().required()
 const requiredStringId = Joi.string().required().max(128)
+const MetaData = Joi.object().optional().description('Arbitrary Metadata').label('MetaData')
 
+const PolicyIdObject = Joi.object({
+  id: Joi.string().required().description('Policy Id'),
+  variables: Joi.object().description('A list of the veriables with their fixed values').label('variables')
+}).required().description('Policy Id Object, alternatively supply string instead').label('PolicyIdObject')
+
+const PolicyIdString = Joi.string().required().description('PolicyIdString')
+// it would be ideal to put the policyid object first, however it causes a swagger doc error
+// as hapi-swagger cannot interpret alternatives
 const requiredPolicy = Joi.alternatives().try([
-  requiredString,
-  Joi.object().description('Policy').keys({
-    id: requiredString.description('Policy Id'),
-    variables: Joi.object().description('A list of the veriables with their fixed values')
-  })
-])
+  PolicyIdString,
+  PolicyIdObject
+]).description('Required Policy (can be object or string)').label('RequiredPolicy')
+
+const PolicyIdsArray = Joi.array().required().items(requiredPolicy).description('can be array of objects with id:string, variables:object properties OR an array of string ids').label('PolicyIdsArray')
+const UsersArray = Joi.array().required().items(requiredString).description('User IDs').label('UsersArray')
+const TeamsArray = Joi.array().required().items(requiredString).description('Teams IDs').label('TeamsArray')
+const ResourcesArray = Joi.array().max(10).items(requiredString.description('A single resource')).single().required().description('A list of Resources').label('ResourcesArray')
+
+const StatementObject = Joi.object({
+  Effect: Joi.string().valid('Allow', 'Deny').label('Effect'),
+  Action: Joi.array().items(Joi.string()).label('Action'),
+  Resource: Joi.array().items(Joi.string()).label('Resource'),
```

Comment:
both action and resource should have a minimum one element

Reply:
do you want me to add for this version?

Reply:
doesn't break tests, done.